### PR TITLE
feat(ctb): Enforce maximum L2 gas limit

### PR DESCRIPTION
**Pull request by clabby**
## Overview

> [!NOTE]
> Blocked by https://github.com/ethereum-optimism/specs/pull/186 being merged.

Enforces a maximum L2 gas limit within the `SystemConfig`. This change helps ensure that OP Stack chain governors keep the L2 block gas limit within a reasonable range in order to guarantee that the L2 blocks may be proven.

### Constants

| Name | Value |
| ----- | ------ |
| `MAX_GAS_LIMIT` | `200_000_000` |

### Security Considerations
* In the `_setResourceConfig` function, the new minimum gas limit is checked to be less than the current `gasLimit`. This value may never be larger than `MAX_GAS_LIMIT`, per the checks in `_setGasLimit`. This ensures that the `minimumGasLimit <= maximumGasLimit`

### CODE REVIEW

**Comment by clabby** [comment]
* **#10501** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/10501?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#10500** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/10500?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @clabby and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by tynes** [comment]
when specs pr

**Comment by clabby** [comment]
> when specs pr

https://github.com/ethereum-optimism/specs/pull/186

File: packages/contracts-bedrock/test/L1/SystemConfig.t.sol [hunk 1]
```diff
@@ -489,7 +505,8 @@ contract SystemConfig_Setters_Test is SystemConfig_Init {
     /// @dev Tests that `setGasLimit` updates the gas limit successfully.
     function testFuzz_setGasLimit_succeeds(uint64 newGasLimit) external {
         uint64 minimumGasLimit = systemConfig.minimumGasLimit();
-        newGasLimit = uint64(bound(uint256(newGasLimit), uint256(minimumGasLimit), uint256(type(uint64).max)));
+        uint64 maximumGasLimit = systemConfig.minimumGasLimit();
```
Comment (mds1) [inline packages/contracts-bedrock/test/L1/SystemConfig.t.sol]:
```suggestion
        uint64 maximumGasLimit = systemConfig.maximumGasLimit();
```

**Review by maurelian — COMMENTED** [review]
Definitely need to fix that test typo with the suggested fix from @mds1, then LGTM.

**Review by maurelian — APPROVED** [review]

**Pull request merged**